File: gpt-3.5/one-shot-with-context/kernels_construct.c
#ifndef T1
//T1:parallel,kernels,construct,V:2.7
int test1(){
    int err = 0;

    real_t * a = (real_t *)malloc(n * sizeof(real_t));
    real_t * b = (real_t *)malloc(n * sizeof(real_t));

    for (int x = 0; x < n; ++x){
        a[x] = 1.0;
        b[x] = 0.0;
    }

    #pragma acc kernels copyout(b[0:n])
    {
        #pragma acc loop independent
        for (int x = 0; x < n; ++x){
            b[x] = a[x];
        }
    }

    for (int x = 0; x < n; ++x){
        if (b[x] != a[x]){
            err += 1;
            break;
        }
    }

    return err;
}

//T2:parallel,kernels,construct,V:2.7
int test2(){
    int err = 0;

    real_t * a = (real_t *)malloc(n * sizeof(real_t));
    real_t * b = (real_t *)malloc(n * sizeof(real_t));
    real_t * c = (real_t *)malloc(n * sizeof(real_t));

    for (int x = 0; x < n; ++x){
        a[x] = 1.0;
        b[x] = 2.0;
        c[x] = 0.0;
    }

    #pragma acc kernels copyout(c[0:n]) copyin(a[0:n],b[0:n])
    {
        #pragma acc loop independent
        for (int x = 0; x < n; ++x){
            c[x] = a[x] + b[x];
        }
    }

    for (int x = 0; x < n; ++x){
        if (c[x] != a[x] + b[x]){
            err += 1;
            break;
        }
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif

#ifndef T2
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test2();
    }
    if (failed != 0){
        failcode = failcode + (1 << 1);
    }
#endif

    return failcode;
}

File: Codellama-34b-Instruct/Template_and_context_final/kernels_construct.f90
#ifndef T1
!T1:kernels construct,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER :: i, j, k
        REAL :: a(100), b(100), c(100)
        REAL :: sum

        !$acc kernels
        !$acc loop gang(static:1)
        DO i = 1, 100
          a(i) = i
          b(i) = i * 2
          c(i) = i * 3
        END DO
        !$acc end kernels

        !$acc kernels
        !$acc loop gang(static:1)
        DO i = 1, 100
          sum = 0
          DO j = 1, 100
            sum = sum + a(j) * b(j) * c(j)
          END DO
          WRITE (*,*) "Sum = ", sum
        END DO
        !$acc end kernels

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM